sqlglot/optimizer/qualify_tables.py

```diff
@@ -60,12 +64,16 @@ def qualify_tables(expression, db=None, catalog=None, schema=None):
                     source = source.replace(
                         alias(
                             source,
-                            name if name else next_name(),
+                            name or source.name or next_name(),
```

Comment:
See example shown in https://github.com/tobymao/sqlglot/pull/1617/files#r1193208207. If the pivot doesn't have an alias, we'll store `''` as a name in the `sources` dict, and so `source.name` here ensures we'll use `tbl` as the alias of the table source.